src/GraphQL/Internal/Syntax/AST.hs

```diff
@@ -53,20 +53,24 @@ import Protolude
 import Test.QuickCheck (Arbitrary(..), listOf, oneof)
 
 import GraphQL.Internal.Arbitrary (arbitraryText)
-import GraphQL.Internal.Name          
+import GraphQL.Internal.Name
   ( Name
   , HasName(..)
   )
-  
+
 -- * Documents
 
 -- | A 'QueryDocument' is something a user might send us.
 --
 -- https://facebook.github.io/graphql/#sec-Language.Query-Document
-newtype QueryDocument = QueryDocument { getDefinitions :: [Definition] } deriving (Eq,Show)
-
-data Definition = DefinitionOperation OperationDefinition
-                | DefinitionFragment  FragmentDefinition
+data QueryDocument = QueryDocument {
+  getDefinitions :: [Definition]
+  , position :: PositionInfo
+  } deriving (Eq,Show)
+
+type PositionInfo = Maybe (Int, Int)
```

Comment:
Quick question: Under what circumstances would position be `Nothing`? Would it make sense to add a brief comment about that?